webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+
+      if (pendingMT != null) {
+        if (!isExpired(pendingMT)) {
+          tmTextUnitRepository
+              .findById(input.getTmTextUnitId())
+              .ifPresent(
+                  tmTextUnit -> {
+                    if (aiTranslationTextUnitFilterService.isTranslatable(tmTextUnit, repository)) {
+                      translateLocales(input, tmTextUnit, repository);
+                    } else {
+                      logger.debug(
+                          "Text unit with name: {} should not be translated, skipping AI translation.",
+                          tmTextUnit.getName());
+                      meterRegistry.counter(
+                          "AITranslateJob.translate.notTranslatable",
+                          Tags.of("repository", repository.getName()));
+                    }
+                  });
+          meterRegistry
+              .timer("AITranslateJob.timeToMT", Tags.of("repository", repository.getName()))
+              .record(Duration.between(JSR310Migration.dateTimeNow(), pendingMT.getCreatedDate()));
+        } else {
+          // If the pending MT is expired, log an error and delete it
+          logger.error("Pending MT for tmTextUnitId: {} is expired", input.getTmTextUnitId());
+          meterRegistry.counter(
+              "AITranslateJob.expired", Tags.of("repository", repository.getName()));
+        }
+      } else {
+        logger.warn("No pending MT found for tmTextUnitId: {}", input.getTmTextUnitId());
+        meterRegistry.counter(
+            "AITranslateJob.noPendingMTFound", Tags.of("repository", repository.getName()));
+      }
+    } catch (Exception e) {
+      logger.error("Error running job for text unit id {}", input.getTmTextUnitId(), e);
+      meterRegistry.counter(
+          "AITranslateJob.error", Tags.of("repository", input.getRepositoryId().toString()));
+    } finally {
+      TmTextUnitPendingMT pendingMT =
+          tmTextUnitPendingMTRepository.findByTmTextUnitId(input.getTmTextUnitId());
+      if (pendingMT != null) {
+        logger.debug("Deleting pending MT for tmTextUnitId: {}", input.getTmTextUnitId());
+        tmTextUnitPendingMTRepository.delete(pendingMT);
+      }
+    }
+    return null;
+  }
+
+  private void translateLocales(
+      AITranslateJobInput input, TMTextUnit tmTextUnit, Repository repository) {
+
+    Map<String, RepositoryLocaleAIPrompt> repositoryLocaleAIPrompts =
+        repositoryLocaleAIPromptRepository
+            .getActiveTranslationPromptsByRepository(input.getRepositoryId())
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    rlap ->
+                        rlap.getLocale() != null
+                            ? rlap.getLocale().getBcp47Tag()
+                            : REPOSITORY_DEFAULT_PROMPT,
+                    Function.identity()));
+    repository.getRepositoryLocales().stream()
+        .filter(rl -> rl.getParentLocale() != null)
```

Comment:
Is this needed to filter out the root locale (i.e.: English in most cases?).

Reply:
Yes, that's filtering out the source language from AI translation